### PR TITLE
Remove the evdev::Error type

### PR DESCRIPTION
**Pull request by coolreader18**
Unnecessary, since all the errors that can happen are just standard OS errors. I was gonna make the `InvalidEvent` condition just be a custom `io::Error`, but I replaced with a panic, since it's more of an function signature thing than like, an IO error.

### CODE REVIEW

File: src/uinput.rs [hunk 1]
```diff
@@ -303,10 +303,12 @@ impl VirtualDevice {
...
-    pub fn process_ff_upload(&mut self, event: UInputEvent) -> Result<FFUploadEvent, Error> {
-        if event.code() != UInputCode::UI_FF_UPLOAD {
-            return Err(Error::InvalidEvent);
-        }
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if `event.code()` is not `UI_FF_UPLOAD`.
+    pub fn process_ff_upload(&mut self, event: UInputEvent) -> io::Result<FFUploadEvent> {
+        assert_eq!(event.code(), UInputCode::UI_FF_UPLOAD);
```
Comment (jeff-hiner) [inline src/uinput.rs:311]: Hmm... if we didn't want to introduce an explicit panic here, we could just return `io::ErrorKind::Other` with a message.

Comment (coolreader18) [reply src/uinput.rs:311]: Yeah, and that's what I did at first, but I feel like this is the kinda thing panics are for - if you passed a value that doesn't make sense to this function, you've made a logic mistake in your program; it's not an error you can pass back up to the user or something.

Comment (jeff-hiner) [reply src/uinput.rs:311]: At that point though, why even have a mandatory parameter that must be a specific value? Maybe we should just remove the param entirely.

Comment (emberian) [reply src/uinput.rs:311]: the other data that comes in on the event is used though, right?

**Review by emberian — APPROVED** [review]

File: src/uinput.rs (hunk 1 above)
Comment (coolreader18) [reply src/uinput.rs:311]: It is, yeah - the value and maybe-but-probably-not the timestamp are relevant.

Comment (jeff-hiner) [reply src/uinput.rs:311]: Hmm... I'm generally more in favor of allowing callers of a library to decide whether or not to panic on something like this, seeing as some folks might be trying to use evdev in embedded apps like kiosks and the like. Would it hurt to return `std::io::Error` with `ErrorKind::Other`?

**Comment by coolreader18** [comment]
Merging as-is - I'll look at ways we could avoid the check altogether, maybe newtypes

**Pull request merged**